src/plugins/plugins.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Utility to register plugins and common namespace for keeping reference to all plugins classes.
+ */
+import { toUpperCaseFirst } from '../helpers/string';
+import { createPriorityQueue } from '../utils/dataStructures/priorityQueue';
+import { createUniqueList } from '../utils/dataStructures/uniqueList';
+import { createUniqueQueue } from '../utils/dataStructures/uniqueQueue';
+
+const ERROR_PLUGIN_REGISTERED = pluginName => `There is already registered "${pluginName}" plugin.`;
+const ERROR_PRIORITY_REGISTERED = priority => `There is already registered plugin on priority "${priority}".`;
+const ERROR_PRIORITY_NAN = priority => `The priority "${priority}" is not a number.`;
+
+/**
+ * Stores plugins names' queue with they priorities.
+ */
+const priorityPluginsQueue = createPriorityQueue({
+  errorPriorityExists: ERROR_PRIORITY_REGISTERED,
+  errorPriorityNaN: ERROR_PRIORITY_NAN,
+});
+/**
+ * Stores plugins names' queue by registration order.
+ */
+const uniquePluginsQueue = createUniqueQueue({
+  errorItemExists: ERROR_PLUGIN_REGISTERED,
+});
+/**
+ * Stores plugins references between theirs' name and class.
+ */
+const uniquePluginsList = createUniqueList({
+  errorIdExists: ERROR_PLUGIN_REGISTERED,
+});
+
+/**
+ * Gets registered plugins' names with an appropriate order.
+ * First ordered plugins by their ascending priority.
+ * Then plugins from the unordered registry in order of registration.
+ *
+ * @returns {string[]}
+ */
+export function getPluginsNames() {
+  return [
+    ...priorityPluginsQueue.getItems(),
+    ...uniquePluginsQueue.getItems(),
+  ];
+}
+/**
+ * Gets registered plugin's class by passed name.
```

Comment:
Hmmmm.... gets or returns? Anyway...
"Returns/Gets registered plugin's class based on the given name".

I get the point (it's about the param) but "by passed name" sounds a little weird to me.
And you got it really nice with "given" just in `registerPlugin` :+1:

Reply:
`Gets registered plugin's class based on the given name` fits great ❤️